src/Shared/FileUtilities.cs

```diff
@@ -1031,27 +1031,80 @@ internal static string MakeRelative(string basePath, string path)
+            string[] splitBase = fullBase.Split(Path.DirectorySeparatorChar);
+            string[] splitPath = fullPath.Split(Path.DirectorySeparatorChar);
 
-            Uri pathUri = CreateUriFromPath(path);
-
-            if (!pathUri.IsAbsoluteUri)
+            if (!path.StartsWith(splitPath[0]))
             {
-                // the path is already a relative url, we will just normalize it...
-                pathUri = new Uri(baseUri, pathUri);
+                // path was already relative so just return it
+                return FixFilePath(path);
             }
 
-            Uri relativeUri = baseUri.MakeRelativeUri(pathUri);
-            string relativePath = Uri.UnescapeDataString(relativeUri.IsAbsoluteUri ? relativeUri.LocalPath : relativeUri.ToString());
-
-            string result = relativePath.Replace(Path.AltDirectorySeparatorChar, Path.DirectorySeparatorChar);
+            int baseI = 0;
+            int pathI = 0;
+            while (true)
+            {
+                if (baseI == splitBase.Length)
+                {
+                    while (String.IsNullOrEmpty(splitPath[pathI]))
+                    {
+                        pathI++;
+                    }
+                    if (pathI == splitPath.Length)
+                    {
+                        return ".";
+                    }
+                    break;
+                }
+                else if (pathI == splitPath.Length)
+                {
+                    break;
+                }
+                else if (String.IsNullOrEmpty(splitBase[baseI]))
+                {
+                    baseI++;
+                }
+                else if (String.IsNullOrEmpty(splitPath[pathI]))
+                {
+                    pathI++;
+                }
+                else if (splitBase[baseI].Equals(splitPath[pathI], PathComparison))
+                {
+                    baseI++;
+                    pathI++;
+                }
+                else
+                {
+                    break;
+                }
+            }
 
-            return result;
+            StringBuilder sb = StringBuilderCache.Acquire();
+            if (baseI == 0)
+            {
+                return fullPath;
+            }
+            while (baseI < splitBase.Length)
+            {
+                if (!String.IsNullOrEmpty(splitBase[baseI]))
+                {
+                    sb.Append("..").Append(Path.DirectorySeparatorChar);
+                }
+                baseI++;
+            }
+            while (pathI < splitPath.Length)
+            {
+                if (!String.IsNullOrEmpty(splitPath[pathI]))
+                {
+                    sb.Append(splitPath[pathI]).Append(Path.DirectorySeparatorChar);
+                }
+                pathI++;
+            }
+            sb.Remove(sb.Length - 1, 1);
+            return sb.ToString();
```

Comment:
```suggestion
            return StringBuilderCache.GetStringAndRelease(sb);
```

Reply:
I was trying to think of that but didn't come up with it. Thanks!